### ISSUE TITLE
XML emitted by adminacl_ajax.php is unescaped; usernames are not validated server-side

### ISSUE THREAD

**Issue by kojiromike** [openemr/openemr#11955]
`library/ajax/adminacl_ajax.php` builds XML responses by string concatenation and emits user-controlled values without escaping. For example, `username_listings_xml()` (line 252) does:

```php
"\t\t<username>" . $iter["username"] . "</username>\n"
```

The same pattern appears in `user_group_listings_xml()`, `error_xml()`, and the `*_listings_xml` family in `src/Common/Acl/AclExtended.php`.

Separately, the user-creation entry points do not validate the `username` value server-side:

- `interface/usergroup/usergroup_admin.php` (Add User form) trims `$_POST['rumple']` and inserts it as-is.
- `src/Validators/PractitionerValidator.php` (used by `POST /api/practitioner` via `PractitionerService::insert()`) has rules for `fname`, `lname`, `npi`, `email` but no rule for `username`.
- `src/Services/FHIR/FhirPractitionerService.php::insertOpenEMRRecord` (used by `POST /fhir/Practitioner`) synthesizes a username via `uniqid($lname . $fname)` from the FHIR HumanName values; those values are not validated for HTML-special characters either.

If a username contains characters like `<` or `>`, the ACL admin screen (`interface/usergroup/adminacl.php`) — which loads via jQuery `dataType: "xml"` — fails to parse the AJAX response and shows "ERROR, unable to collect data from server" until that user is removed or deactivated. Reaching the entry points requires the `admin/users` ACL (or `admin/super` for the CDA import path), so this isn't a privilege-boundary issue, but it's a real correctness/availability bug and worth fixing as defense in depth.

### Suggested scope
- XML-escape every value written by the `*_listings_xml` and `error_xml` functions in `library/ajax/adminacl_ajax.php` and `src/Common/Acl/AclExtended.php`.
- Reject (or normalize) HTML-special and control characters in `username` at every server-side entry point — the form handler, `PractitionerValidator`, and the FHIR parser path.
